### DISCUSSION TITLE
Scroll beyond last line

### DISCUSSION THREAD

**Discussion by johnbillion** [standardnotes/forum#2625] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**

Yes!

**Is your feature request related to a problem? Please describe.**

When I'm working on a note and the cursor reaches the bottom of the visible window, it always feels like you're typing up against a barrier because the cursor buts up against the bottom of the window.

**Describe the solution you'd like**

I'd like to be able to scroll beyond the bottom of the last line, so I can add text to the bottom of a note without having to do so with my cursor right at the bottom of the window.

This feature is becoming more common in code editors (such VS Code) and publishing tools (such as WordPress).

**Describe alternatives you've considered**

The Joplin app provides this feature. My approach in Standard Notes when I really want to get my cursor away from the bottom of the window is to insert a bunch of empty lines at the end.

**Screenshot**

Without scroll-past-end:

<img alt="" width="400" src="https://user-images.githubusercontent.com/208434/84801996-368e3780-b000-11ea-9ac3-3e36212d9a7d.png">

With scroll-past-end:

<img width="400" alt="" src="https://user-images.githubusercontent.com/208434/84802189-84a33b00-b000-11ea-8346-67ff9a0de46c.png">

**Comment by JaspalSuri** [comment]
Hi John, thanks for the feedback! I understand why this feature would be useful. I'll pass it along to the team for review!

**Comment by virtuallyvlad** [comment]
Yeah, without this feature the editor feels immature and usability is compromised, as you'll always be staring at the bottom of the window when writing longer notes.

I'd also add that it should allow you to scroll down far enough that the last line would be at the very top of the window/viewport, because some may prefer the input line or cursor to be at the very top.

You can do a hacky workaround like adding blank lines after the last line of text, to let your last line of text appear higher in the window/frame, but that's more annoying when writing/editing and should really be taken care of by the editor.

This is sometimes called "Scroll Past End", "Scroll Beyond Last Line", or "Virtual Space".  A related but different feature is "Typewriter scrolling", which always keeps your cursor centered in the viewport as you type.

This applies to the desktop app (tested on Windows), web app (tested in Chrome), and mobile app (tested on iOS).

Similar:
https://github.com/standardnotes/forum/issues/581

**Comment by jasonwaleryszak** [comment]
Is there a status on this enhancement? It's something you completely forget about when its there, but reminded of constantly when it is not.
